src/lib/services/version-service.ts

```diff
@@ -194,25 +216,29 @@ export default class VersionService {
                 };
 
                 if (this.telemetryEnabled) {
-                    const featureInfo = await this.getFeatureUsageInfo();
-                    versionPayload.featureInfo = featureInfo;
+                    versionPayload.featureInfo =
+                        await this.getFeatureUsageInfo();
                 }
-                const res = await fetch(this.versionCheckUrl, {
-                    method: 'POST',
-                    body: JSON.stringify(versionPayload),
-                    headers: { 'Content-Type': 'application/json' },
-                });
-                if (res.ok) {
-                    const data = (await res.json()) as IVersionResponse;
-                    this.latest = {
-                        oss: data.versions.oss,
-                        enterprise: data.versions.enterprise,
-                    };
-                    this.isLatest = data.latest;
+                if (this.versionCheckUrl) {
```

Comment:
I think this change is unrelated but it's worth having it independently of this PR